pim-tool-back/src/main/java/vn/elca/training/service/impl/ProjectServiceImpl.java

```diff
@@ -1,30 +1,89 @@
 package vn.elca.training.service.impl;
 
-import org.springframework.context.annotation.Profile;
+import com.querydsl.jpa.impl.JPAQuery;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
 import vn.elca.training.model.entity.Project;
-import vn.elca.training.repository.ProjectRepository;
+import vn.elca.training.model.entity.QProject;
+import vn.elca.training.model.exception.ProjectNumberAlreadyExistsException;
 import vn.elca.training.service.ProjectService;
 
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
 import java.util.List;
 
-/**
- * @author vlp
- *
- */
 @Service
-@Profile("!dummy | dev")
 public class ProjectServiceImpl implements ProjectService {
-
     private ProjectRepository projectRepository;
+    @PersistenceContext
+    private EntityManager em;
+
+    public ProjectServiceImpl(ProjectRepository projectRepository) {
+        this.projectRepository = projectRepository;
+    }
+
+    @Override
+    public String welcome() {
+        return "Welcome to ELCA Bootcamp";
+    }
+
+    @Override
+    public Project addProject(Project project) {
+        if (checkProjectNumber(project.getProjectNumber())) {
+            throw new ProjectNumberAlreadyExistsException("The project number already existed. Please select a different number");
+        } else {
+            return projectRepository.save(project);
```

Comment:
The else clause is redundant, it can be simplify as follow.
```
if (checkProjectNumber(project.getProjectNumber())) {
            throw new ProjectNumberAlreadyExistsException("The project number already existed. Please select a different number");

return projectRepository.save(project);
```